docs/references/events.mdx

```diff
@@ -562,3 +562,85 @@ This [Event](/workflows#event) type indicates that a [Workflow Execution](/workf
+This event is added to the caller's workflow history for Asynchronous Nexus Operations, for example one that is picked up by a handler's SDK client and started by a Nexus handler.
+The event is not added to the caller's Workflow history for Synchronous Nexus Operations, since they transition directly to [NexusOperationCompleted](#nexusoperationcompleted) or another final state such as [NexusOperationFailed](#nexusoperationfailed) when the response is provided synchronously by the Nexus handler.
+
+| Field              | Description                                                                                                                                 |
+| :----------------- | :------------------------------------------------------------------------------------------------------------------------------------------ |
+| scheduled_event_id | The ID of the [NexusOperationScheduled](#nexusoperationscheduled) event this task corresponds to.                                           |
+| operation_id       | The operation ID returned by the Nexus handler in the response to the StartOperation request. This ID is used when canceling the operation. |
+| request_id         | The request ID allocated at schedule time.                                                                                                  |
+
+### NexusOperationCompleted
+
+This Event type indicates that the Nexus Operation has completed successfully.
+The caller's workflow history records the result of a successful Nexus Operation with this event for synchronous and asynchronous Nexus Operations.
+This Event type contains Nexus Operation Execution results.
+
+| Field              | Description                                                                                                                                                          |
+| :----------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| scheduled_event_id | The ID of the [NexusOperationScheduled](#nexusoperationscheduled)` event. Uniquely identifies this operation.                                                        |
+| result             | Serialized result of the Nexus operation. The response of the Nexus handler. Delivered either via a completion callback or as a response to a synchronous operation. |
+| request_id         | The request ID allocated at schedule time.                                                                                                                           |
+
+### NexusOperationFailed
+
+This Event type indicates that the Nexus Operation has failed.
+The caller's workflow history records a failed Nexus Operation with this event both for synchronous and asynchronous Nexus Operations.
+For example, when an SDK client picks up a Nexus Operation and the Nexus handler responds synchronously with a non-retryable error.
```

Comment:
SDK client is a wrong term here. This may be a bit more readable.

```suggestion
For example, when a Nexus Handler responds synchronously with a non-retryable error or when a Workflow  that backs an Operation fails, resulting in a [WorkflowExecutionFailed](#workflowexecutionfailed) event.
```